docs/src/manual/constraints.md

```diff
@@ -55,4 +56,23 @@ apply!(K, f, ch)
 a = K\f
 apply!(a, ch) # enforces affine constraints
 
-```
\ No newline at end of file
+```
+
+Note: Replacing `f` with `r=-f` and solving `a = -K\r` in the example above may give the wrong result.
+Instead, use the procedure for nonlinear problems when applying affine constraints to such problem formulations.
+
+### Solving nonlinear problems
+A Newton-type solution method for the nonlinear system `r(a)=0` uses the 
+update `Δa=-K\r` (where `K=∂r/∂a` if the standard Newton's method is used).
```

Comment:
I would add explicitly here something like
```
It is important to check the residual **after** applying boundary conditions so that residual entries for constrained dofs are handled properly. The following pseudo-code shows the typical pattern for solving a non-lienar problem with Newtons method: